mifospay/src/main/java/org/mifos/mobilewallet/mifospay/passcode/ui/PassCodeActivity.java

```diff
@@ -59,10 +60,15 @@ public void startNextActivity() {
         // authenticate user with saved Preferences
         mPresenter.createAuthenticatedService();
 
-        Intent intent = new Intent(PassCodeActivity.this, MainActivity.class);
-        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP |
-                Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
-        startActivity(intent);
+        if ("editProfile".equals(getIntent().getStringExtra("editProfile"))) {
```

Comment:
Then in this line of code do something like
 `if(getIntent().getStringExtra(Constants.PASSCODE_NEXT_ACTIVITY)
                    .equals(EditProfileActivity.class.getSimpleName())){`

Reply:
Using this causes the app to crash as seen in logcat it shows NullException error so instead I used
**if(EditProfileActivity.class.getSimpleName()).equals(getIntent().getStringExtra(Constants.PASSCODE_NEXT_ACTIVITY))** and it works fine.